src/docmanager/index.ts

```diff
@@ -0,0 +1,438 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IContentsModel, IKernelId, IKernelSpecId, IContentsOpts, IKernel,
+  INotebookSession
+} from 'jupyter-js-services';
+
+import {
+  IDisposable, DisposableDelegate
+} from 'phosphor-disposable';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+
+/**
+ * The interface for a document model.
+ */
+export interface IDocumentModel {
+  /**
+   * A signal emitted when the document content changes.
+   */
+  contentChanged: ISignal<IDocumentModel, any>;
+
+  /**
+   * Serialize the model.  It should return a JSON object or a string.
+   */
+  serialize(): any;
+
+  /**
+   * Deserialize the model from a string or a JSON object.
+   *
+   * #### Notes
+   * Should emit a [contentChanged] signal.
+   */
+  deserialize(value: any): void;
+
+  /**
+   * The default kernel name for the the document.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  defaultKernelName: string;
+
+  /**
+   * The default kernel language for the document.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  defaultKernelLanguage: string;
+}
+
+
+/**
+ * A session info object for running sessions.
+ */
+export interface ISessionInfo {
+  /**
+   * The list of file paths associated with the running sessions.
+   */
+  paths: string[];
+
+  /**
+   * The list of kernel specs associated with the running sessions.
+   */
+  specs: IKernelSpecId[];
+}
+
+
+/**
+ * The document context object.
+ */
+export interface IDocumentContext {
+  /**
+   * A signal emitted when the kernel changes.
+   */
+  kernelChanged: ISignal<IDocumentContext, IKernel>;
+
+  /**
+   * A signal emitted when the path changes.
+   */
+  pathChanged: ISignal<IDocumentContext, string>;
+
+  /**
+   * The current kernel associated with the document.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  kernel: IKernel;
+
+  /**
+   * The current path associated with the document.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  path: string;
+
+  /**
+   * Change the current kernel associated with the document.
+   */
+  changeKernel(options: IKernelId): Promise<IKernel>;
+
+  /**
+   * Save the the document contents to disk.
+   */
+  save(): Promise<void>;
+
+  /**
+   * Revert the document contents to disk contents.
+   */
+  revert(): Promise<void>;
+
+  /**
+   * Get the list of running sessions.
+   */
+  listRunningSessions(): Promise<ISessionInfo>;
```

Comment:
listActiveSessions? or maybe just listSessions? or is this just the sessions that contain active kernels?